sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/functions.scala

```diff
@@ -930,3 +930,327 @@ object HyperLogLogPlusPlus {
   )
   // scalastyle:on
 }
+
+/**
+ * A central moment is the expected value of a specified power of the deviation of a random
+ * variable from the mean. Central moments are often used to characterize the properties of about
+ * the shape of a distribution.
```

Comment:
This is not for the case when `n = 0` or `n =1` but when we have `null` or `NaN` in the values. Are we ignoring them or outputting `NaN` directly?